From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <lynxplay101@gmail.com>
Date: Fri, 19 Jul 2024 19:09:21 +0200
Subject: [PATCH] Move CraftComplexRecipe to extend CraftingRecipe

A craft complex recipe wraps a CustomRecipe, which itself is a
CraftingRecipe.
As such, this complex recipe should also be a crafting recipe.

diff --git a/src/main/java/org/bukkit/inventory/CraftingRecipe.java b/src/main/java/org/bukkit/inventory/CraftingRecipe.java
index 9575e40c7ef3e1d86d10d83b33d9e68f0ed7406f..e8c3afda92d4ae5430d622ea18500985d6cc00f2 100644
--- a/src/main/java/org/bukkit/inventory/CraftingRecipe.java
+++ b/src/main/java/org/bukkit/inventory/CraftingRecipe.java
@@ -12,8 +12,11 @@ import org.jetbrains.annotations.NotNull;
  * Represents a shaped or shapeless crafting recipe.
  */
 public abstract class CraftingRecipe implements Recipe, Keyed {
-    private final NamespacedKey key;
-    private final ItemStack output;
+    // Paper - make CraftComplexRecipe extend CraftingRecipe - start
+    protected NamespacedKey key;
+    protected ItemStack output;
+    protected CraftingRecipe() {}
+    // Paper - make CraftComplexRecipe extend CraftingRecipe - end
     private String group = "";
     private CraftingBookCategory category = CraftingBookCategory.MISC;
 
